### PR TITLE
feat (#59): UsersRepository implementation

### LINKED ISSUE #59: [Users] Create and implement `UsersRepository` using timemates/sdk with caching locally in database
> **Note**
> Take a look at [sqldelight first](https://cashapp.github.io/sqldelight/2.0.0/), if you want to implement cache within this task.
>
> **Should be done as soon as possible.**

[Contract of repository](https://github.com/timemates/app/blob/04d84039af5dc58d20c980198663b40e670c1ff1/feature/users/domain/src/commonMain/kotlin/io/timemates/app/users/repositories/UsersRepository.kt) is already declared, you should only implement it in `data` module.

### PR DESCRIPTION
**Pull request by kotleni**
**Is it breaking change?**
No.

**Specify linked issues or discussions**

Close #59 

**Additional information**

Implemented `UsersRepository`, `CachedUsersDataSource` and `CachedUsers` sql. 
Changed in interface `UserRepository`, and classes `GetUserUseCase`, `GetUsersUseCase`.

Now, if repository have cache - this data will be emited to flow first. 
And after - the data will be loaded from the SDK and will also be emited and updated in cache.

### CODE REVIEW

File: feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class CachedUsersDataSource(
+    private val cachedUsersQueries: CachedUsersQueries
+) {
+    init {
+        // TODO: Move it to other place
+        CoroutineScope(Dispatchers.Default).launch {
```
Comment (y9vad9) [inline feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt:20]: зміни або на GlobalScope, або передавай CoroutineScope через конструктор

Comment (y9vad9) [reply feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt:20]: також, я думаю, що має клірити відповідно до часу, наприклад кеш актуальний 5 хв.

File: feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/UsersRepository.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            }
+            .getOrElse { throwable -> Result.failure(throwable) }
+            .let {
+                emit(it)
+                it.onSuccess { user -> cachedUsersDataSource.saveUser(user) }
+            }
+    }
+
+    override suspend fun getUsers(ids: List<UserId>): Flow<Result<List<User>>> = flow {
+        // Emit local saved data (is have)
```
Comment (y9vad9) [inline feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/UsersRepository.kt:37]:
```suggestion
        // Emit local saved data (if have)
```

File: feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt (hunk 1 above)
Comment (y9vad9) [inline feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt:20]:
```suggestion
        GlobalScope.launch(Dispatchers.Default) {
```

**Review by y9vad9 — CHANGES_REQUESTED** [review]

File: feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+class CachedUsersDataSource(
+    private val cachedUsersQueries: CachedUsersQueries,
+    private val coroutineScope: CoroutineScope
+) {
+    init {
+        // TODO: Move it to other place
+        coroutineScope.launch { clear() }
+    }
+
+    private suspend fun clear() = withContext(Dispatchers.IO) {
```
Comment (y9vad9) [inline feature/users/data/src/commonMain/kotlin/io/timemates/app/users/data/CachedUsersDataSource.kt:27]: I think we still should clear cache by time, not by creating it. User can close the app and reopen it much faster. I think cache should serve ~10 minutes.

**Review by 1Rinkana — APPROVED** [review]

**Pull request merged**